HandsUp/Assets/CardViewScene/Scripts/Category/CategoryManager.cs

```diff
@@ -0,0 +1,180 @@
+    private PlayerManager playerManager;
+    public GameObject categoryItem;
+
+    private List<Category> categories;
+    private List<CustomCategory> customCategories;
+
+
+    private void Awake()
+    {
+        playerManager = GameObject.Find("PlayerManager").GetComponent<PlayerManager>();
+        categories = new List<Category>();
+        customCategories = new List<CustomCategory>();
+    }
+
+
+    public void InitCategories()
+    {
+        GameObject.Find("Canvas").transform.Find("CategoriesScrollView").gameObject.SetActive(true);
+        GameObject.Find("Canvas").transform.Find("CardsScrollView").gameObject.SetActive(false);
+
+        if (categories.Count <= 0)
+        {
+            GetBuitInCategoriesFromServer();
+        }
+
+        if (customCategories.Count > 0)
+        {
+            GameObject[] content = GameObject.FindGameObjectsWithTag("customCategoryItem");
+            for (int i = 0; i < content.Length; i++)
+            {
+                Destroy(content[i]);
+            }
+            customCategories.Clear();
+        }
+
+        if (playerManager.GetUserId() >= 0)
+        {
+            GetCustomCategoriesFromServer(playerManager.GetUserId());
+        }
+    }
+
+
+    private void GetBuitInCategoriesFromServer()
+    {
+            StartCoroutine(DataManager.getDataFromServer("category", (raw) =>
+           {
+               Debug.Log(raw);
+               JObject applyJObj = JObject.Parse(raw);
+               if (applyJObj["result"].ToString().Equals("success"))
+               {
+                   foreach (JObject tmpCategory in applyJObj["categories"])
+                   {
+                       Category tmp = new Category();
+                       tmp.SetId((int)tmpCategory["category_id"]);
+                       tmp.SetName(tmpCategory["category_name"].ToString());
+
+                       categories.Add(tmp);
+                   }
+                   CreateNewCategoryItems(categories);
+               }
+               else
+               {
+                   Debug.Log("results : fail");
+               }
+
+           }));
+    }
+
+
+    private void CreateNewCategoryItems(List<Category> categories)
+    {
+        for (int i = 0; i < categories.Count; i++)
+        {
+            GameObject newCategoryItem = Instantiate(categoryItem, new Vector3(0, 0, 0), Quaternion.identity);
+            newCategoryItem.transform.SetParent(GameObject.Find("Canvas").transform.Find("CategoriesScrollView/Viewport/Content").transform);
+            Debug.Log(newCategoryItem.transform.parent.name);
+            newCategoryItem.transform.localScale = new Vector3(1, 1, 1);
+            newCategoryItem.GetComponent<Category>().SetId(categories[i].GetId());
+            newCategoryItem.GetComponent<Category>().SetName(categories[i].GetName());
+
+            newCategoryItem.GetComponentInChildren<Text>().text = categories[i].GetName();
+        }
+    }
+
+
+    public List<Category> GetBuitInCategories()
```

Comment:
이 부분 오타 수정해야할 것 같습니다!
```suggestion
    public List<Category> GetBuiltInCategories()
```

Reply:
넵 수정했습니다